client/webserver/site/src/js/coinexplorers.ts

```diff
@@ -0,0 +1,154 @@
+    [Mainnet]: (cid: string) => `https://blockexplorer.one/dash/mainnet/tx/${cid.split(':')[0]}`,
+    [Testnet]: (cid: string) => `https://blockexplorer.one/dash/testnet/tx/${cid.split(':')[0]}`,
+    [Simnet]: (cid: string) => `https://blockexplorer.one/dash/mainnet/tx/${cid.split(':')[0]}`
+  },
+  133: { // zec
+    [Mainnet]: (cid: string) => `https://zcashblockexplorer.com/transactions/${cid.split(':')[0]}`,
+    [Testnet]: (cid: string) => `https://blockexplorer.one/zcash/testnet/tx/${cid.split(':')[0]}`,
+    [Simnet]: (cid: string) => `https://zcashblockexplorer.com/transactions/${cid.split(':')[0]}`
+  },
+  147: { // zcl
+    [Mainnet]: (cid: string) => `https://explorer.zcl.zelcore.io/tx/${cid.split(':')[0]}`,
+    [Simnet]: (cid: string) => `https://explorer.zcl.zelcore.io/tx/${cid.split(':')[0]}`
+  },
+  136: { // firo
+    [Mainnet]: (cid: string) => `https://explorer.firo.org/tx/${cid.split(':')[0]}`,
+    [Testnet]: (cid: string) => `https://testexplorer.firo.org/tx/${cid.split(':')[0]}`,
+    [Simnet]: (cid: string) => `https://explorer.firo.org/tx/${cid.split(':')[0]}`
+  },
+  145: { // bch
+    [Mainnet]: (cid: string) => `https://bch.loping.net/tx/${cid.split(':')[0]}`,
+    [Testnet]: (cid: string) => `https://tbch4.loping.net/tx/${cid.split(':')[0]}`,
+    [Simnet]: (cid: string) => `https://bch.loping.net/tx/${cid.split(':')[0]}`
+  },
+  966: { // matic
+    [Mainnet]: (cid: string) => {
+      const [arg, isAddr] = ethBasedExplorerArg(cid)
+      return isAddr ? `https://polygonscan.com/address/${arg}` : `https://polygonscan.com/tx/${arg}`
+    },
+    [Testnet]: (cid: string) => {
+      const [arg, isAddr] = ethBasedExplorerArg(cid)
+      return isAddr ? `https://mumbai.polygonscan.com/address/${arg}` : `https://mumbai.polygonscan.com/tx/${arg}`
+    },
+    [Simnet]: (cid: string) => {
+      const [arg, isAddr] = ethBasedExplorerArg(cid)
+      return isAddr ? `https://polygonscan.com/address/${arg}` : `https://polygonscan.com/tx/${arg}`
+    }
+  }
+}
+
+export function formatCoinID (cid: string) {
+  if (cid.startsWith(coinIDTakerFoundMakerRedemption)) {
+    const makerAddr = cid.substring(coinIDTakerFoundMakerRedemption.length)
+    return intl.prep(intl.ID_TAKER_FOUND_MAKER_REDEMPTION, { makerAddr: makerAddr })
+  }
+  return cid
+}
+
+/*
+ * baseChainID returns the asset ID for the asset's parent if the asset is a
+ * token, otherwise the ID for the asset itself.
+ */
+function baseChainID (assetID: number) {
+  const asset = app().user.assets[assetID]
+  return asset.token ? asset.token.parentID : assetID
+}
+
+/*
+ * setCoinHref sets the hyperlink element's href attribute based on provided
+ * assetID and data-explorer-coin value present on supplied link element.
+ */
+export function setCoinHref (assetID: number, link: PageElement) {
+  // setCoinHref may get called from `insertRichNote()` earlier than the user
+  // data is fetched, therefore we need to check if the user
+  // object is available.
```

Comment:
This is no longer true. I'd let it error if we use this before we have a `user` now.